authors/apps/articles/favourites_views.py

```diff
@@ -0,0 +1,91 @@
+import json
+
+from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+from django.shortcuts import get_object_or_404, get_list_or_404
+from django.urls import reverse
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from rest_framework import status
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import (
+    AllowAny,
+    IsAuthenticated,
+    IsAuthenticatedOrReadOnly
+)
+from rest_framework.response import Response
+
+from config.settings import default
+from .models import Favourites, Article
+from .pagination import ArticleOffsetPagination
+from .renderers import FavortiesJsonRenderer
+from .serializers import FavouriteSerializer
+
+from authors.apps.profiles.models import Profile
+from authors.apps.authentication.models import User
+
+
+class FavouritesView(GenericAPIView):
+    serializer_class = FavouriteSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (FavortiesJsonRenderer,)
+
+    def get_object(self, **kwargs):
+        article_slug = self.kwargs.get('slug')
+        article_obj = get_object_or_404(Article, slug=article_slug)
+        return article_obj
+
+    def put(self, request, **kwargs):
+        article = self.get_object()
+        user = request.user
+        article_id = article.id
+        fav = Favourites.objects.filter(
+            article_id=article_id, user=user
+        ).first()
+        if fav:
```

Comment:
@Rhytah , could you refactor this logic a bit and also create a helper method which is documented